tests/unit/test__init__.py

```diff
@@ -83,3 +83,62 @@ def test_check_python_version_does_not_warn_on_supported(mock_version_tuple):
 
         # Assert that warnings.warn was NOT called
         mock_warn_call.assert_not_called()
+
+
+def test_determine_all_includes_json_when_available():
+    """
+    Test that _determine_all includes JSON types when both are truthy.
+    """
+
+    from db_dtypes import _determine_all
+
+    # Simulate available types (can be any truthy object)
+    mock_json_array = object()
+    mock_json_dtype = object()
+
+    result = _determine_all(mock_json_array, mock_json_dtype)
+
+    expected_all = [
+        "__version__",
+        "DateArray",
+        "DateDtype",
+        "TimeArray",
+        "TimeDtype",
+        "JSONDtype",
+        "JSONArray",
+        "JSONArrowType",
+    ]
+    assert set(result) == set(expected_all)
+    assert "JSONDtype" in result
+    assert "JSONArray" in result
+    assert "JSONArrowType" in result
```

Comment:
Nit: these don't seem necessary if we already verified that the sets are equal.

Reply:
Removed.